### PR TITLE
vendor: update to containerd v1.6.0-beta.4

### PR DESCRIPTION
**Pull request by egernst**
Update our containerd vendoring. In particular, we're interested in
grabbing the updated annotation definitions for defining sandbox sizing.

- go get github.com/containerd/containerd@v1.6.0-beta.4
- edit go.mod to remove containerd v1.5.8 replacement directive
- go mod vendor
- go mod tidy

Fixes: #3276

Of particular interest for our usage:

```
 
 src/runtime/vendor/github.com/containerd/containerd/pkg/cri/annotations/annotations.go 

 @@ -32,6 +32,16 @@ const (
 	// SandboxID is the sandbox ID annotation
 	SandboxID = "io.kubernetes.cri.sandbox-id"

 	// SandboxCPU annotations are based on the initial CPU configuration for the sandbox. This is calculated as the
 	// sum of container CPU resources, optionally provided by Kubelet (introduced  in 1.23) as part of the PodSandboxConfig
 	SandboxCPUPeriod = "io.kubernetes.cri.sandbox-cpu-period"
 	SandboxCPUQuota  = "io.kubernetes.cri.sandbox-cpu-quota"
 	SandboxCPUShares = "io.kubernetes.cri.sandbox-cpu-shares"

 	// SandboxMemory is the initial amount of memory associated with this sandbox. This is calculated as the sum
 	// of container memory, optionally provided by Kubelet (introduced in 1.23) as part of the PodSandboxConfig.
 	SandboxMem = "io.kubernetes.cri.sandbox-memory"
```

Signed-off-by: Eric Ernst <eric_ernst@apple.com>

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
The bump itself looks good and it also brings in a newer runc, which fixes a dependabot alert.
Thanks, @egernst!

**Review by fengwang666 — APPROVED** [review]

**Review by liubin — APPROVED** [review]

**Comment by bergwolf** [comment]
CI failed at
```
./hack/tree_status.sh
tree is dirty, please commit all changes

 M src/runtime/vendor/modules.txt
diff --git a/src/runtime/vendor/modules.txt b/src/runtime/vendor/modules.txt
index d6cd6231..2f331d1c 100644
--- a/src/runtime/vendor/modules.txt
+++ b/src/runtime/vendor/modules.txt
@@ -283,8 +283,6 @@ github.com/shurcooL/sanitized_anchor_name
 ## explicit
 github.com/sirupsen/logrus
 github.com/sirupsen/logrus/hooks/syslog
-# github.com/smartystreets/goconvey v1.6.4
make[1]: *** [Makefile:659: vendor] Error 1
make: *** [Makefile:31: vendor-runtime] Error 2
-## explicit
 # github.com/stretchr/testify v1.7.0
 ## explicit
 github.com/stretchr/testify/assert
make[1]: Leaving directory '/home/runner/work/kata-containers/kata-containers/src/github.com/kata-containers/kata-containers/src/runtime'
Error: The operation was canceled.
```
Please include modules.txt change in your commit.

File: src/runtime/go.mod [hunk 1]
```diff
@@ -6,62 +6,59 @@ require (
...
-	k8s.io/apimachinery v0.20.6
-	k8s.io/cri-api v0.20.6
+	google.golang.org/grpc v1.41.0
+	k8s.io/apimachinery v0.22.0
+	k8s.io/cri-api v0.23.0-alpha.4
 )
 
 replace (
-	github.com/containerd/containerd => github.com/containerd/containerd v1.5.8
 	github.com/opencontainers/runc => github.com/opencontainers/runc v1.0.1
```
Comment (Jakob-Naucke) [inline src/runtime/go.mod:62]: I think this replace should be removed.

**Comment by jodh-intel** [comment]
Ping @egernst.

**Comment by egernst** [comment]
back from vacation!  Addressed modules.txt, re-pushed...

/test

**Pull request merged**